### PR TITLE
ynh_install_nodejs

### PR DESCRIPTION
**Pull request by yalh76**
Tested on 
- https://github.com/YunoHost-Apps/wikijs_ynh/pull/137

### CODE REVIEW

File: ynh_install_nodejs/ynh_install_nodejs [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        ynh_print_info --message="Removing of nodenv-"$nodenv_version
+        ynh_secure_remove --file="$nodenv_install_dir"
+        ynh_secure_remove --file="/etc/profile.d/nodenv.sh"
+        
+        # Remove previous n Version Management
+        n_install_dir="/opt/node_n"
+        ynh_secure_remove --file="$n_install_dir"
+        ynh_secure_remove --file="/usr/local/n"
+        sed --in-place "/$n_install_dir/d" /root/.bashrc
+        rm --force /etc/cron.daily/node_update
```
Comment (kay0u) [inline ynh_install_nodejs/ynh_install_nodejs]:
```suggestion
        ynh_secure_remove --file="/etc/cron.daily/node_update"
```

File: ynh_install_nodejs/ynh_install_nodejs [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+#
+# Don't forget to execute Node.js-dependent command in a login environment
+# (e.g. sudo --login option)
+# When not possible (e.g. in systemd service definition), please use direct path
+# to nodenv shims (e.g. $NODENV_ROOT/shims/bundle)
+#
+# usage: ynh_install_nodejs --nodejs_version=nodejs_version
+# | arg: -v, --nodejs_version= - Version of Node.js to install.
+#
+# Requires YunoHost version 2.7.12 or higher.
```
Comment (kay0u) [inline ynh_install_nodejs/ynh_install_nodejs]: Getops, used in this helper, requiers YunoHost 3.2.2
https://github.com/YunoHost/yunohost/blob/adc83b4c9c2c30e9ef75f3609c538b646f91f1db/data/helpers.d/getopts#L47

File: ynh_install_nodejs/ynh_install_nodejs [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    # Move an existing Node.js binary, to avoid to block nodenv
+    test -x /usr/bin/node && mv /usr/bin/node /usr/bin/node_n
+    test -x /usr/bin/npm && mv /usr/bin/npm /usr/bin/npm_n
+
+    # Install or update nodenv
+    nodenv="$(command -v nodenv $nodenv_install_dir/bin/nodenv | head -1)"
+    if [ -n "$nodenv" ]; then
+        ynh_print_info --message="nodenv already seems installed in \`$nodenv'."
+        pushd "${nodenv%/*/*}"
+            if git remote -v 2>/dev/null | grep "https://github.com/nodenv/nodenv.git"; then
```
Comment (kay0u) [inline ynh_install_nodejs/ynh_install_nodejs:123]: How do you handle the case where this git remote doesn't exist?

**Review by kay0u — APPROVED** [review]
:+1:

File: ynh_install_nodejs/ynh_install_nodejs (hunk 2 above)
Comment (yalh76) [reply ynh_install_nodejs/ynh_install_nodejs]: updated :)

File: ynh_install_nodejs/ynh_install_nodejs (hunk 3 above)
Comment (yalh76) [reply ynh_install_nodejs/ynh_install_nodejs:123]: it's not managed :/

But wasn't managed on original helper: 
https://github.com/YunoHost/yunohost/blob/adc83b4c9c2c30e9ef75f3609c538b646f91f1db/data/helpers.d/nodejs#L16-L27

You think it's mandatory to manage it ?

**Comment by yalh76** [comment]
Switched to https://github.com/YunoHost/yunohost/pull/1196

**Pull request closed without merging**